src/test/ui/async-await/async-await.rs

```diff
@@ -70,6 +70,13 @@ fn async_nonmove_block(x: u8) -> impl Future<Output = u8> {
     }
 }
 
+fn async_closure(x: u8) -> impl Future<Output = u8> {
```

Comment:
This bit already exists in `async-closure.rs` so it doesn't need to be tested again (the separation is also because it's good to test the feature gates in isolation).

However, it would be good to extend `async-closure.rs` with something similar to what you have below,
e.g. `unsafe { async move |x: u8| unsafe_fn(unsafe_async_fn(x).await) }`.

Reply:
done (d023e47877d28ffdaeb586c50d1271970f7c84f4 + 87006a9031afa81d4e102e3cb7d06d0852e1a22f)

Reply:
and fixed in 66381efdec6b85f8e626f69e4a6fcc53616b40b9